src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -136,14 +149,16 @@ public void CompleteStream(bool errored)
                     }
                 }
 
-                _http2Output.Dispose();
+                _http2Output.Complete();
 
                 RequestBodyPipe.Reader.Complete();
 
                 // The app can no longer read any more of the request body, so return any bytes that weren't read to the
                 // connection's flow-control window.
                 _inputFlowControl.Abort();
 
+                CanReuse = !_keepAlive && _requestProcessingStatus == RequestProcessingStatus.ResponseCompleted;
```

Comment:
Should you be able to use errored here instead for setting CanReuse?

Reply:
Stephen wanted to check these fields. A response that hasn't finished might have a task that is still running.

I've added a comment.